azurerm/internal/services/apimanagement/tests/api_management_custom_domain_resource_test.go

```diff
@@ -0,0 +1,177 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMApiManagementCustomDomain_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_custom_domain", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMApiManagementCustomDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagementCustomDomain_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementCustomDomainExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMApiManagementCustomDomain_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_custom_domain", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMApiManagementCustomDomainDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagementCustomDomain_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementCustomDomainExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMApiManagementCustomDomain_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMApiManagementCustomDomainDestroy(s *terraform.State) error {
+	conn := acceptance.AzureProvider.Meta().(*clients.Client).ApiManagement.ServiceClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_api_management_custom_domain" {
+			continue
+		}
+
+		serviceName := rs.Primary.Attributes["api_management_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(ctx, resourceGroup, serviceName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				// TODO: Is this an error case? If Custom Domains is destroyed, it should not destroy the API Management Service, so what are we testing for here?
+				// If the entire configuration is torn down, including the API Management Service, then it, too, will not be found, and returning nil is correct.
+				return nil
```

Comment:
Since the CheckDestroy func is run after a destroy, the APIM service is probably gone too, so it's fine to return nil here.

```suggestion
				return nil
```